### DISCUSSION TITLE
Using Snowflake database through PDO?

### DISCUSSION THREAD

**Discussion by wesyah234** [ADOdb/ADOdb#1052] [Q&A] [answered]
Can someone shed some light on Snowflake support?  I've used ADOdb for many years with oracle, mysql, and postgresql.  Now I need to connect to a snowflake db. I was able to get a working connection using PDO directly (my sysadmin installed pdo_snowflake) 

`~> php -m | grep pdo_

pdo_mysql

pdo_pgsql

pdo_snowflake

pdo_sqlite
`

so I then came to ADOdb to see if there was a class/driver included for this.  There is no mention of snowflake on the ADOdb site.  I also checked the 2 pages for PDO, and neither of them mention snowflake either.  (It's not one of the supported drivers on this page https://adodb.org/dokuwiki/doku.php?id=v5:database:pdo) .. So, I decided to try it anyway, and what do you know, it worked great:

'
$dsn = 'snowflake:account=xyzabc;database=123456;schema=schema123';
$username = 'username';
$password = 'xyz';
$conn = ADONewConnection("pdo");
$conn->Connect($dsn, $username, $password);
$sql = "SELECT * FROM table";
$users = $conn->Execute("$sql");
`

So, is this totally expected?  I don't use PDO at all since I've been using ADOdb all my life, so just not sure how all this is coming together... ... should ADODB mention that it works with snowflake, or should people just know that if they can get something working with PDO directly, that it will automatically then work with ADODB using the 'pdo' driver type..?

Thanks!

**Comment by mnewnham** [comment, marked as answer]
As I understand it, **_pdo_snoflake__** is a third-party plugin built on top of the PHP PDO library. On a personal note, I must admit to having never heard of it before.

Having said that, because it is built on top of the standard PDO library, it's not unreasonable to assume that the basic functionality of the driver _**execute() etc**_ would work correctly. However, its highly likely that some of the functionality that is dependent on database-specific interrogations, such as **meta()** functions are unlikely to work. That is where the  ADOdb driver specific functions come into play.

If you want to build a "proper" ADOdb pdo snowflike driver, its easier to build a complete driver by cloning an existing one. Is snowflake built as fork of some other database that has true PDO support?

Whilst we are able to provide guidance on building out a driver, I'm not sure we can ourselves provide any time at the moment unless you want us to do it under the umbrella of billed professional services.

**Comment by wesyah234** [comment]
Ok, thank you.  Your response prompted me to look a bit further into what my sysadmin installed.  Seems it was this:

https://github.com/snowflakedb/pdo_snowflake

I guess for now I'll just go ahead and start using it and see how far I get running queries through ADOdb.  I generally just run CRUD statements through ADOdb so I am unlikely to run into any issues you mentioned.

Thanks!
